clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -984,6 +984,8 @@ public void close(final Timer timer) {
             }
         } finally {
             super.close(timer);
+            // Super-class close may wait for more commit callbacks to complete.
+            invokeCompletedOffsetCommitCallbacks();
```

Comment:
Agree, there could be async requests, with known coord, not getting a response in the above while loop, then getting it while the super.close waits, so we trigger the callbacks.  But this makes me notice, aren't we breaking the close(Duration) contract here, calling that super.close(timer) on the finally clause? Let's say async requests that are not getting a response within the timeout in the above while loop (so we block for time on the while), then `finally`, the super class blocks for that time again [here](https://github.com/apache/kafka/blob/8b274d8c1bfbfa6d4319ded884a11da790d7bf77/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java#L1139) . Am I missing something? (I can file a separate Jira if I'm not missing something here)

Reply:
That timer will be expired, right? So when we wait inside finally, we will not block again in `awaitPendingRequests` in the super-class.

Reply:
You're right! (silly me, I was probably thinking about a duration arg instead of a timer), all good then.